app/controllers/applications_controller.rb

```diff
@@ -0,0 +1,53 @@
+class ApplicationsController < ApplicationController
+
+  def index
+    @applications = Application.all
+  end
+  
+  def new
+    @application = Application.new
+  end
+
+  def create
+    @application = Application.new(application_params)
+    if @application.save
+      redirect_to "/applications/#{@application.id}"   
+    else
+      render "new"
+    end
+  end
+
+  def show
+    @application = Application.find(params[:id])
+    @pets = @application.pets
+  end
+
+  def search_pets
+    @application = Application.find(params[:id])
+    @search_results = @application.search_pets_by_name(params[:name])
+    render 'show'
+  end
+
+  def submit
+    @application = Application.find(params[:id])
+    if @application.update(application_params.merge(application_status: 'Pending'))
+      redirect_to "/applications/#{@application.id}"
+    else
+      render 'show'
+    end
+  end
+
+  def adopt_pet
+    application = Application.find(params[:id])
```

Comment:
If we're thinking about CRUD actions, what's happening here is a `create`, as we are creating a joins record (PetApplication) for a given pet and application. However, we wouldn't want to put this create action in this controller because we already have one above. 

So, a better location for this code would be a joins controller - consider making a PetApplicationsController where you could have a create action that does this implementation: creates a join record for a given pet and app